controllers/cloudstackmachine_controller.go

```diff
@@ -186,6 +187,7 @@ func (r *CloudStackMachineReconciliationRunner) DeleteMachineIfFailuredomainNotE
 // Implicitly it also fetches its bootstrap secret in order to create said instance.
 func (r *CloudStackMachineReconciliationRunner) GetOrCreateVMInstance() (retRes ctrl.Result, reterr error) {
 	if r.CAPIMachine.Spec.Bootstrap.DataSecretName == nil {
+		r.Recorder.Event(r.ReconciliationSubject, "Normal", "Creating", "Bootstrap DataSecretName not yet available.")
 		return r.RequeueWithMessage("Bootstrap DataSecretName not yet available.")
```

Comment:
should we instead refactor RequeueWithMessage to always write an event to the Reconciliation subject?

Reply:
requeueWithMessage can potentially be used by other controller (not capc machine controller), it may make the scope of this PR. When other controllers use requeueWithMessage, it may need attach events to different resources.